### ISSUE TITLE
AttributeError: '_io.StringIO' object has no attribute 'buffer'

### ISSUE THREAD

**Issue by candicegjing** [codalab/codalab-worksheets#1957]
```
Error on request by matt.downey18(0xd72e93b8f2b448fc9318d136e89e555a):

POST /cli/command

Query params:

Form params:

JSON body:
{
"command": "cl info 0xdcf10e1da32e45ce970bbb07fd4694eb -v",
"worksheet_uuid": "0x6756cd59b87145a5bc031cde9c66d8ca"
}

Local variables:
contents=<codalab.worker.file_util.un_gzip_stream.<locals>.UnGzipStream object at 0x7f678015a7f0>
kwargs={'head': 50, 'tail': 50, 'truncation_text': '\n... truncated ...\n\n'}
info_type='file'
info={'name': 'stderr', 'size': 0, 'perm': 420, 'type': 'file'}
tail=None
head=10
subpath='stderr'
bundle_uuid='0xdcf10e1da32e45ce970bbb07fd4694eb'
client=<codalab.client.json_api_client.JsonApiClient object at 0x7f675c3f4358>
self=<codalab.lib.bundle_cli.BundleCLI object at 0x7f675c3f4e80>

Traceback (most recent call last):
File "/opt/codalab-worksheets/codalab/server/rest_server.py", line 106, in wrapper
return callback(*args, **kwargs)
File "/usr/local/bin/bottle.py", line 1732, in wrapper
rv = callback(*a, **ka)
File "/opt/codalab-worksheets/codalab/server/json_api_plugin.py", line 18, in wrapper
result = callback(*args, **kwargs)
File "/opt/codalab-worksheets/codalab/rest/cli.py", line 48, in post_worksheets_command
return general_command(query['worksheet_uuid'], query['command'])
File "/opt/codalab-worksheets/codalab/rest/cli.py", line 149, in general_command
structured_result = cli.do_command(args)
File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 839, in do_command
structured_result = command_fn()
File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 833, in <lambda>
command_fn = lambda: args.function(self, args)
File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 2088, in do_info_command
self.print_contents(client, info)
File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 2175, in print_contents
self.print_target_info(client, bundle_uuid, item['name'], head=10)
File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 2318, in print_target_info
shutil.copyfileobj(contents, self.stdout.buffer)
AttributeError: '_io.StringIO' object has no attribute 'buffer'
```

**Comment by candicegjing** [comment]
Might be related to https://github.com/pexpect/pexpect/issues/30

**Comment by candicegjing** [comment]
Also reproduced locally `cl info <bundle_id> -v`, don't see any issue.

**Comment by epicfaace** [comment]
I had initially added this code in the python 2 -> 3 migration. I needed to use "shutil.copyfileobj" to copy a _binary_ file to stdout, so I had to use `this.stdout.buffer` (which corresponds to `sys.stdout.buffer` when `this.stdout` = `sys.stdout`).

From the [Python 3 documentation](https://docs.python.org/3/library/sys.html)
> Note To write or read binary data from/to the standard streams, use the underlying binary buffer object. For example, to write bytes to stdout, use sys.stdout.buffer.write(b'abc').
However, if you are writing a library (and do not control in which context its code will be executed), be aware that the standard streams may be replaced with file-like objects like io.StringIO which do not support the buffer attribute.

And this time, `sys.stdout` is a `StringIO` instance, which doesn't support the `buffer` attribute:

https://github.com/codalab/codalab-worksheets/blob/5b9390b5ecdead2ef736c4e43dc94037f2aeb161/codalab/rest/cli.py#L97
